docs/configuration.rst

```diff
@@ -317,6 +318,183 @@ set the ``tub.location`` option described below.
+
+In the future, Tahoe services may be changed to live on HTTP/HTTPS URLs
+instead of Foolscap. In that case, connections will be made using whatever
+handler is configured for ``tcp`` hints. So the same ``tcp = tor``
+configuration will work.
+
+``[tor]``
+---------
+
+This controls how Tor connections are made. The defaults (all empty) mean
+that, when Tor is needed, the node will try to connect to a Tor daemon's
+SOCKS proxy on localhost port 9050 or 9150. Port 9050 is the default Tor
+SOCKS port, so it should be available under any system Tor instance (e.g. the
+one launched at boot time when the standard Debian ``tor`` package is
+installed). Port 9150 is the SOCKS port for the Tor Browser Bundle, so it
+will be available any time the TBB is running.
+
+You can set ``launch = True`` to cause the Tahoe node to launch a new Tor
+daemon when it starts up (and kill it at shutdown), if you don't have a
+system-wide instance available. Note that it takes 30-60 seconds for Tor to
+get running, so using a long-running Tor process may enable a faster startup.
+If your Tor executable doesn't live on ``$PATH``, use ``tor.executable=`` to
+specify it.
+
+``[tor]``
+
+``enable = (boolean, optional, defaults to True)``
+
+    If False, this will disable the use of Tor entirely. The default of True
+    means the node will use Tor, if necessary, and if possible.
+
+``socks.port = (string, optional, PORT, defaults to empty)``
+
+    This tells the node that Tor connections should be routed to a SOCKS
+    proxy listening on the given port. The default (of an empty value) will
+    cause the node to first try localhost port 9050, then if that fails, try
+    localhost port 9150. These are the default listening ports of the
+    standard Tor daemon, and the Tor Browser Bundle, respectively.
+
+``control.port = (string, optional, endpoint specification string)``
+
+    This tells the node to connect to a pre-existing Tor daemon on the given
+    control port (which is typically ``unix://var/run/tor/control`` or
+    ``tcp:localhost:9051``). The node will then ask Tor what SOCKS port it is
+    using, and route Tor connections to that.
+
+``launch = (bool, optional, defaults to False)``
+
+    If True, the node will spawn a new (private) copy of Tor at startup, and
+    will kill it at shutdown. The new Tor will be given a persistent state
+    directory under ``NODEDIR/private/``, where Tor's microdescriptors will
+    be cached, to speed up subsequent startup.
+
+``tor.executable = (string, optional, defaults to empty)``
+
+    This controls which Tor executable is used when ``launch = True``. If
+    empty, the first executable program named ``tor`` found on ``$PATH`` will
+    be used.
+
+There are 5 valid combinations of these configuration settings:
+
+* 1: ``(empty)``: use SOCKS on port 9050/9150
+* 2: ``launch = true``: launch a new Tor
+* 3: ``socks.port = HOST:PORT``: use an existing Tor on the given SOCKS port
```

Comment:
both the socks port and the control port should be twisted endpoints so that the user can possibly specify a unix domain socket